### PR TITLE
 Standard template for CHECKMULTISIG with 17~20 keys

### PR DESCRIPTION
**Pull request by jl2012**
Making CHECKMULTISIG with 17~20 keys standard. Trying to redo #11545 after #13194 is merged

### CODE REVIEW

**Comment by sipa** [comment]
I think this should be conditional on the environment (and only enabled inside SegWit scripts).

**Comment by fivepiece** [comment]
Out of interest, does the soft fork nature of it means that `{17,18,19} checkmultisig(verify)` will still count as 20 sigops (`MAX_PUBKEYS_PER_MULTISIG`) ?
As in here :
https://github.com/bitcoin/bitcoin/blob/fd96d54f39cf4f66890e0bb40812d47e69728cec/src/script/script.cpp#L161-L167

**Comment by jl2012** [comment]
@sipa, I think legacy CHECKMULTISIG is always restricted to up to 3 keys, and P2SH is restricted by MAX_P2SH_SIGOPS

**Comment by sipa** [comment]
@jl2012 I'm aware, but I'm worried about situations were Solver is called on a script that isn't found in a transaction that already passed validation.

**Comment by jl2012** [comment]
oh.....actually, arbitrary scripts are already standard in witness v0....so this is only for wallet?

**Comment by instagibbs** [comment]
Yes they're already standard. The only thing preventing their use was wallet code.

File: src/script/standard.cpp [hunk 1]
```diff
@@ -57,10 +57,18 @@ static bool MatchPayToPubkeyHash(const CScript& script, valtype& pubkeyhash)
     return false;
 }
 
-/** Test for "small positive integer" script opcodes - OP_1 through OP_16. */
-static constexpr bool IsSmallInteger(opcodetype opcode)
+/** Test for "small positive integer" script opcodes - OP_1 through OP_16, or a single byte push for 17 to 20. */
```
Comment (instagibbs) [inline src/script/standard.cpp]: The old definition is used elsewhere still. Perhaps rename the function to "IsCheckMultisigInteger" or something more direct?

Comment (jl2012) [reply src/script/standard.cpp]: I think it's only used here, and is newly introduced in  #13194

**Comment by jl2012** [comment]
@fivepiece, no, this has nothing to do with sigop counting. They are still counted as 20

File: src/script/standard.cpp (hunk 1 above)
Comment (instagibbs) [reply src/script/standard.cpp]: I meant the term "small integer" i.e. "encode small integer" etc

**Comment by jl2012** [comment]
Closed in favour of #13449

**Pull request closed without merging**